common/helpers/field.js

```diff
@@ -177,14 +179,58 @@ function insertItemConditional({ key, field }) {
   }
 }
 
+function findParent(field) {
+  if (Object.prototype.hasOwnProperty.call(field, 'items')) {
+    return true
+  }
+}
+
+async function populateParentItems({ name }) {
+  const assessmentQuestions = await referenceDataService.getAssessmentQuestions(
+    name
+  )
+
+  return assessmentQuestions
+    .filter(referenceDataHelpers.filterDisabled())
+    .map(mapAssessmentQuestionToConditionalField)
+    .map(mapAssessmentQuestionToTranslation)
+    .map(mapReferenceDataToOption)
+}
+
+function setChildValidation(key, field, parentField) {
+  if (!field.validate) {
+    return
+  }
+
+  const conditionalParent = parentField.items.find(item =>
+    key.includes(item.key)
+  )
+
+  field.dependent = {
+    field: parentField.name,
+    value: conditionalParent.value,
+  }
+}
+
+async function setupAssessmentQuestions(fields) {
+  const parentField = Object.values(fields).find(findParent)
```

Comment:
I wonder if there's a little too much abstraction here which feels like it makes it a little harder to follow. Lots of this only originally occurred in one place so not sure it warrants abstracting everything out into shared functions as they're never reused anywhere.

I think the new terms `parent`/`child` threw me a little too as its another term we're using. Wonder  if we can maybe make use more of `dependent` which is the term already being used when one field's validation depends on another's. It doesn't necessarily need to therefore be a parent/child relationship.

Reply:
Wonder if we can use something more like `hasItems` as opposed to calling them parent/child too?